File: 287-find-the-duplicate-number/287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        for(int i=0; i<nums.size(); i++){
            int valAtInx = abs(nums[i]);
            if(nums[valAtInx] < 0) return valAtInx;
            else nums[valAtInx]*=-1;
        }
        return 0;
    }
};

File: 287-find-the-duplicate-number/287-find-the-duplicate-number.java
class Solution {
    public int abs(int n){
        if(n<0) return n*-1;
        return n;
    }
    public int findDuplicate(int[] nums) {
        int len = nums.length;
        for(int i=0; i<len; i++){
            int val = abs(nums[i]);
            if(nums[val]<0) return val;
            nums[val]=nums[val]*-1;
            
        }
        return 0;
    }
}